diff --git a/src/xml/parsers/lv_xml_label_parser.c b/src/xml/parsers/lv_xml_label_parser.c
index d1ae650cb..868893d90 100644
--- a/src/xml/parsers/lv_xml_label_parser.c
+++ b/src/xml/parsers/lv_xml_label_parser.c
@@ -61,6 +61,7 @@ void lv_xml_label_apply(lv_xml_parser_state_t * state, const char ** attrs)
         else if(lv_streq("translation_tag", name)) lv_label_set_translation_tag(item, value);
 #endif
         else if(lv_streq("bind_text", name)) {
+            if(value == NULL || value[0] == '\0') continue;
             lv_subject_t * subject = lv_xml_get_subject(&state->scope, value);
             if(subject == NULL) {
                 LV_LOG_WARN("Subject \"%s\" doesn't exist in label bind_text", value);
