diff --git a/src/widgets/slider/lv_slider.c b/src/widgets/slider/lv_slider.c
index f1ec507d2..4d6da73e8 100644
--- a/src/widgets/slider/lv_slider.c
+++ b/src/widgets/slider/lv_slider.c
@@ -327,8 +327,9 @@ static void lv_slider_event(const lv_obj_class_t * class_p, lv_event_t * e)
             }
         }
         else if(indev_type == LV_INDEV_TYPE_POINTER) {
-            if(is_slider_horizontal(obj)) lv_obj_add_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_VER);
-            else  lv_obj_add_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_HOR);
+            /* Block perpendicular scroll while dragging (better touchscreen UX) */
+            if(is_slider_horizontal(obj)) lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_VER);
+            else  lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_HOR);
         }
     }
     else if(code == LV_EVENT_FOCUSED) {
@@ -338,13 +339,12 @@ static void lv_slider_event(const lv_obj_class_t * class_p, lv_event_t * e)
         }
     }
     else if(code == LV_EVENT_SIZE_CHANGED) {
+        /* Block perpendicular scroll chain (better touchscreen UX) */
         if(is_slider_horizontal(obj)) {
-            lv_obj_add_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_VER);
-            lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_HOR);
+            lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_VER);
         }
         else {
-            lv_obj_add_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_HOR);
-            lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_VER);
+            lv_obj_remove_flag(obj, LV_OBJ_FLAG_SCROLL_CHAIN_HOR);
         }
         lv_obj_refresh_ext_draw_size(obj);
     }
